### DISCUSSION TITLE
Transactions Error and No Personal Access Token

### DISCUSSION THREAD

**Discussion by bmgaspard** [firefly-iii/firefly-iii#6187] [Firefly III] [answered]
I recently installed Firefly III on a hosted server using CPanel app installer. When I try to add transactions, I get the error "Internal Firefly III Exception: Unable to read key from file file:///home4/fuelbget/public_html/storage/oauth-public.key" I also can't create a Personal access token, I click create new token, give it a name, but when I click create nothing happens.

I am very new to Firefly III so if there is any specific info you need, please let me know.

**Comment by JC5** [comment, marked as answer]
The installation routine of Firefly III needs write access to the storage directory. You may need to reinstall it. Otherwise, I can't really help you, these Softaculous installation scripts are not maintained by me.

**Reply by bmgaspard** [reply]
Thanks! I'll work on it to see what I can do. If I can't get it working I'll move it to a docker.
